pymongo/cursor.py

```diff
@@ -131,33 +142,27 @@ class CursorType:
 class _SocketManager:
     """Used with exhaust cursors to ensure the socket is returned."""
 
-    def __init__(self, sock, more_to_come):
+    def __init__(self, sock: SocketInfo, more_to_come: bool):
         self.sock = sock
```

Comment:
We should instead make this `Optional[None]` instead of adding the ignore assignment so the other methods can handle when it is `None`

Reply:
If I make it `Optional[SockInfo]`, then we'd have to add an ignore on line 158 because we'd try to call `unpin` on something that can be `None`. Is that preferred?

Reply:
Perhaps "if not self.closed" can be removed and replaced with `if self.sock`?

Reply:
I still think `Optional[SockInfo]` is the right type to use (haha on my typo earlier).

Reply:
yeah, so now that the if statement doesn't rely on `close`, sock is `Optional[SockInfo]`

Reply:
To be clear, the parameter should still be `SockInfo`, but the property should be `Optional[SockInfo]`

Reply:
ahh I see I see, sorry about that. Fixed it now!